sdk-lib/src/main/java/cash/z/ecc/android/sdk/block/CompactBlockProcessor.kt

```diff
@@ -1345,27 +1340,17 @@ class CompactBlockProcessor internal constructor(
     }
 
     suspend fun calculateBirthdayHeight(): BlockHeight {
-        var oldestTransactionHeight: BlockHeight? = null
-        @Suppress("TooGenericExceptionCaught")
-        try {
-            val tempOldestTransactionHeight = repository.getOldestTransaction()?.minedHeight
-                ?: lowerBoundHeight
-            // to be safe adjust for reorgs (and generally a little cushion is good for privacy)
-            // so we round down to the nearest 100 and then subtract 100 to ensure that the result is always at least
-            // 100 blocks away
-            oldestTransactionHeight = BlockHeight.new(
-                network,
-                tempOldestTransactionHeight.value -
-                    tempOldestTransactionHeight.value.rem(MAX_REORG_SIZE) - MAX_REORG_SIZE.toLong()
-            )
-        } catch (t: Throwable) {
-            Twig.debug(t) { "failed to calculate birthday" }
-        }
-        return buildList<BlockHeight> {
-            add(lowerBoundHeight)
-            add(backend.network.saplingActivationHeight)
-            oldestTransactionHeight?.let { add(it) }
-        }.maxOf { it }
+        return repository.getOldestTransaction()?.minedHeight?.value?.let {
+            // To be safe adjust for reorgs (and generally a little cushion is good for privacy), so we round down to
+            // the nearest 100 and then subtract 100 to ensure that the result is always at least 100 blocks away
+            var oldestTransactionHeightValue = it
+            oldestTransactionHeightValue -= oldestTransactionHeightValue.rem(MAX_REORG_SIZE) - MAX_REORG_SIZE.toLong()
+            if (oldestTransactionHeightValue < lowerBoundHeight.value) {
+                lowerBoundHeight
+            } else {
+                BlockHeight.new(network, oldestTransactionHeightValue)
+            }
+        } ?: lowerBoundHeight
```

Comment:
I refactored the whole `calculateBirthdayHeight` method to provide more clarity and rid of the unnecessary and too generic exception catching. The `lowerBoundHeight` is already compared with `network.saplingActivationHeight` at the top of the class, so there is no need for a second `maxOf`.